src/escape.d

```diff
@@ -76,28 +169,36 @@ bool checkEscape(Scope* sc, Expression e, bool gag)
 
         override void visit(AddrExp e)
         {
-            result |= checkEscapeRef(sc, e.e1, gag);
+            if (checkEscapeRef(sc, e.e1, true))
+                result = e;
         }
 
         override void visit(SymOffExp e)
         {
-            check(e.loc, e.var);
+            VarDeclaration v = e.var.isVarDeclaration();
+            if (v && v.toParent2() == sc.func)
+            {
+                if (v.isDataseg())
+                    return;
+                if ((v.storage_class & (STCref | STCout)) == 0)
+                    result = e;
+            }
         }
 
         override void visit(VarExp e)
         {
             VarDeclaration v = e.var.isVarDeclaration();
             if (v)
```

Comment:
The win scales from short ones to long ones.

"No need to visit stuff that's not a variable declaration" is grokked in two lines instead of 23. Mind your readers.